framework/include/vectorpostprocessors/LineMaterialSamplerBase.h

```diff
@@ -138,19 +138,16 @@ LineMaterialSamplerBase<T>::execute()
   std::vector<Elem *> intersected_elems;
   std::vector<LineSegment> segments;
 
-  MooseSharedPointer<PointLocatorBase> plb = MooseSharedPointer<PointLocatorBase>(_fe_problem.mesh().getMesh().sub_point_locator().release());
-
-  Moose::elementsIntersectedByLine(_start, _end, _fe_problem.mesh(), plb, intersected_elems, segments);
+  UniquePtr<PointLocatorBase> pl = _mesh.getPointLocator();
+  Moose::elementsIntersectedByLine(_start, _end, _mesh, pl, intersected_elems, segments);
```

Comment:
@friedmud this looked a bit fishy to me (by maybe you can clarify). You obtained the intersected elements from the undisplaced mesh, but below use the element pointers from that mesh to initialize the (potentially) _displaced_ mesh. I've changed this to consistently use the `_subproblem.mesh()`. Let me know if that is correct.